### ISSUE TITLE
"ELI5" for people without technology engagement

### ISSUE THREAD

**Issue by shleeable** [fedi-e2ee/public-key-directory-specification#71]
PKI and E2E messaging which this enables is excellent... but I'd be keen to resolve the PGP problem by trying our best to make sure this service is usable by people (who are unable or unwilling to have more than a basic technological depth).

Can this service be used by people who don't know they need it? Can we offer this by default/passively in a way that will never require intervention from the user?

or is there always going to be a barrier to entry, even if it is low...  Now low can we make it?

**Comment by soatok** [comment]
> Can this service be used by people who don't know they need it?

Yes, absolutely.

So, the development plan from here looks something like this:

1. Write and review the hell out of a specification. **(We are here!)**
2. Develop a reference implementation.
3. Go through painstaking technical steps to make sure it's actually secure.
4. Write client software (SDKs, etc.) for developers to use.

The SDKs will have an API that looks something like:

```js
const validPublicKeys = await PKD.getKeys('soatok@furry.engineer');
const getAuxData = await PKD.getAuxData('soatok@furry.engineer', 'ssh');
```

It must be simple!

Once we have SDKs in several languages with the most reasonable APIs possible, we will then move onto developing the E2EE project atop these SDKs.

Client-side software will focus on usability and low friction. Users don't need to know any of this technical mumbo jumbo. They'll just install (or, in most cases, update their existing) client software and go through a (hopefully easy-to-follow) enrollment process (which is mostly going to be "do you want power user features? how do you want to backup your secret keys?" etc.).

Unless they are interested in technology, my hope is that everything I'm doing in this project is totally invisible to them. They don't need to even care about it.

**Comment by soatok** [comment]
https://github.com/fedi-e2ee/public-key-directory-specification/pull/69/commits/f33d1213e04b8db490fc11f81ce2c9dc27d07b7f should make this intent more explicit. (That PR is updating the readme.)
